homeassistant/components/roborock/__init__.py

```diff
@@ -43,13 +47,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         for device in device_map.values()
     }
     network_results = await asyncio.gather(
-        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients.values())
+        *(mqtt_client.get_networking() for mqtt_client in mqtt_clients.values()),
+        return_exceptions=True,
     )
-    network_info = {
-        device.duid: result
-        for device, result in zip(device_map.values(), network_results)
-        if result is not None
-    }
+    network_info = {}
+    for device, result in zip(device_map.values(), network_results):
+        if result is not None and not isinstance(result, RoborockException):
+            network_info[device.duid] = result
+        else:
+            _LOGGER.warning(
+                "Failed to connect to get networking information about %s", device.duid
+            )
+            if isinstance(result, RoborockException):
+                _LOGGER.exception(result)
```

Comment:
Why do we only log RoborockException?

Reply:
Based off of a conversation here: 

https://github.com/home-assistant/core/pull/94209#discussion_r1253869212

I can change it to add exception if you would like